gradio/components/rich_textbox.py

```diff
@@ -0,0 +1,295 @@
+        Event listener for when the user selects text in the Textbox.
+        Uses event data gradio.SelectData to carry `value` referring to selected substring, and `index` tuple referring to selected range endpoints.
+        See EventData documentation on how to use this event data.
+        """
+        IOComponent.__init__(
+            self,
+            label=label,
+            info=info,
+            every=every,
+            show_label=show_label,
+            container=container,
+            scale=scale,
+            min_width=min_width,
+            interactive=interactive,
+            visible=visible,
+            elem_id=elem_id,
+            elem_classes=elem_classes,
+            value=value,
+            **kwargs,
+        )
+        self.type = type
+        self.rtl = rtl
+        self.text_align = text_align
+
+    def get_config(self):
+        return {
+            "lines": self.lines,
+            "max_lines": self.max_lines,
+            "placeholder": self.placeholder,
+            "value": self.value,
+            "type": self.type,
+            "autofocus": self.autofocus,
+            "show_copy_button": self.show_copy_button,
+            "container": self.container,
+            "text_align": self.text_align,
+            "rtl": self.rtl,
+            "file_count": self.file_count,
+            "file_types": self.file_types,
+            **IOComponent.get_config(self),
+        }
+
+    @staticmethod
+    def update(
+        value: dict[str, str | list[str]]
+        | Literal[_Keywords.NO_VALUE]
+        | None = _Keywords.NO_VALUE,
+        lines: int | None = None,
+        max_lines: int | None = None,
+        placeholder: str | None = None,
+        label: str | None = None,
+        info: str | None = None,
+        show_label: bool | None = None,
+        container: bool | None = None,
+        scale: int | None = None,
+        min_width: int | None = None,
+        visible: bool | None = None,
+        interactive: bool | None = None,
+        text_align: Literal["left", "right"] | None = None,
+        rtl: bool | None = None,
+        show_copy_button: bool | None = None,
+        autofocus: bool | None = None,
+    ):
+        return {
+            "lines": lines,
+            "max_lines": max_lines,
+            "placeholder": placeholder,
+            "label": label,
+            "info": info,
+            "show_label": show_label,
+            "container": container,
+            "scale": scale,
+            "min_width": min_width,
+            "visible": visible,
+            "value": value,
+            "interactive": interactive,
+            "show_copy_button": show_copy_button,
+            "autofocus": autofocus,
+            "text_align": text_align,
+            "rtl": rtl,
+            "__type__": "update",
+        }
+
+    def preprocess(
+        self, x: dict[str, str | list[dict[str, Any]]] | None
+    ) -> (
+        dict[
+            str,
+            str
+            | bytes
```

Comment:
Why not just always return a list even if it has one element? Would make the types nicer and there's no need to modify the event handler function in your demo if you switch between single and multiple upload